src/mpi/comm/comm_split_type.c

```diff
@@ -109,6 +109,11 @@ int MPIR_Comm_split_type_node(MPIR_Comm * user_comm_ptr, int split_type, int key
 
 #ifdef HAVE_HWLOC
 
+struct info_obj {
+    hwloc_obj_type_t obj_type;
+    char *device_id;
+};
```

Comment:
Do we really need this structure?  Can't we simply pass the `device_id` as a parameter to the `node_split_io` function instead?  Btw, the `device_id` should be a `const char *` since it's not being modified once it's passed to that function.

Reply:
I'll change this to arguments to a function instead.

Reply:
I'm not sure how that helps.  The second field of the structure is a pointer, not the actual device ID string.  So you are comparing the pointers across different processes.